src/Commands/Component.php

```diff
@@ -0,0 +1,12 @@
+<?php
+namespace Airwire\Commands;
+
+use Illuminate\Support\Str;
+
+class Component {
+    public static function register(string $name)
+    {
+        $service_provider = base_path() . "\app\Providers\AppServiceProvider.php";
+        file_put_contents($service_provider, preg_replace('/register[(][)][\s\S].*/', "register() {\n\t\tAirwire::component('" . Str::snake($name) . "', {$name}::class);", file_get_contents($service_provider)));
```

Comment:
The `register()` method is used for registering things into the service container, while the `boot()` method is used for booting the registered services with data.

In this case, `register()` does happen to work well, because `Airwire` is a static class. But if it weren't, this wouldn't work.

So it's better to use `boot()`.

Next, the regex can be replaced with a simple `str_replace()`.

The code should look something like this:

```php
try {
    $path = app_path('Providers/AppServiceProvider.php');
    
    $snake = Str::snake($name);
    
    file_put_contents($path, str_replace(
        'function boot()\n    {',
        <<<PHP
        function boot()
        {
            \Airwire\Airwire::component('{$snake}', {$name}::class);
        PHP,
        file_get_contents($path)
    ));
} catch (Exception $exception) // import Exception at the top
{
    $this->error('The component could not be registered. Please check your app/Providers/AppServiceProvider.php');
}
```